e2e/tests/transfer/base_test.go

```diff
@@ -105,6 +105,16 @@ func (s *TransferTestSuite) TestMsgTransfer_Succeeds_Nonincentivized() {
 		s.Require().Equal(expected, actualBalance.Int64())
 	})
 
+	if testvalues.TokenMetadataFeatureReleases.IsSupported(chainBVersion) {
+		t.Run("metadata for token exists on chainB", func(t *testing.T) {
+			balances, err := chainB.AllBalances(ctx, chainBAddress, true)
+			s.Require().NoError(err)
+
+			// balance for IBC token returns a human-readable denomination
+			s.Require().Equal(chainBIBCToken.GetFullDenomPath(), balances[1].Denom)
+		})
+	}
+
```

Comment:
ah it looks like we have been doing some overlapping work. I have a similar thing in [this PR](https://github.com/cosmos/ibc-go/pull/4649/files#diff-3db9c852a84acf6fb611b95d58dab0086f53f6880f32c0b561799ca26c689708R107) however I did not know we already had `TokenMetadataFeatureReleases!

Reply:
Ah, I just added it for this test. We can fix the merge conflicts later.